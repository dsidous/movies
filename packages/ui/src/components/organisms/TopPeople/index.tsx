import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import Skeleton from '@material-ui/lab/Skeleton';

import MyPager from '../../atoms/Pager';
import Section from '../../atoms/Section';
import MediaImage from '../../atoms/MediaImage';
import PageTransition from '../../atoms/PageTransition/index';
import SEO from '../../atoms/SEO';
import useStyles from './styles';
import { Top_person } from '@typesRoots/toppeople';

interface Props {
  toppeople: Top_person[];
  page: number;
  handlePageSelect: (e: number) => void;
}

const TopPeopleProfile: React.FC<Props> = ({
  toppeople,
  page,
  handlePageSelect,
}) => {
  const classes = useStyles();

  return (
    <PageTransition>
      <SEO title='Popular people' />
      <Section noborder disableGutters>
        <Section.Header>POPULAR PEOPLE</Section.Header>
        <div className={classes.list}>
          {(!toppeople.length ? Array.from(new Array(8)) : toppeople).map(
            (person, index) => (
              <Fragment key={person?.id || index}>
                {toppeople.length ? (
                  <Link key={person.id} to={`/person/${person.id}`}>
                    <figure className={classes.item}>
                      <MediaImage
                        mediaType='profile'
                        size={1}
                        filePath={person.profile_path}
                        name={person.name}
                      />
                      <figcaption className={classes.name}>
                        {person.name}
                      </figcaption>
                    </figure>
                  </Link>
                ) : (
                  <div>
                    <Skeleton variant='rect' height={300} />
                    <Skeleton />
                  </div>
                )}
              </Fragment>
            )
          )}
        </div>
        <MyPager page={page} handlePageSelect={handlePageSelect} />
      </Section>
    </PageTransition>
  );
};

export default TopPeopleProfile;
