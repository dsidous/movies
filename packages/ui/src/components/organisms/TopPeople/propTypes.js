import { number, array, func } from 'prop-types';

export const propTypes = {
  page: number.isRequired,
  toppeople: array.isRequired,
  handlePageSelect: func.isRequired,
};

export const defaultProps = {};
