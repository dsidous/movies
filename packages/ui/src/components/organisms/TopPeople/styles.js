import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    list: {
      display: 'grid',
      gridGap: theme.spacing(3),
      gridTemplateColumns: '1fr 1fr',
      textAlign: 'center',

      [theme.breakpoints.up('sm')]: {
        gridGap: theme.spacing(5),
        gridTemplateColumns: 'repeat(4, 1fr)',
      },
    },

    item: {
      cursor: 'pointer',
      margin: 0,
      overflow: 'hidden',

      '& img': {
        borderRadius: theme.shape.borderRadius,
        boxShadow: '0 0 20px 5px rgba(0, 0, 0, 0.1)',
        transition: 'all 0.3s ease-in',
        width: '100%',
      },

      '&:hover img': {
        transform: 'scale(1.035)',
      },

      '&:hover $name': {
        color: '#000',
      },
    },

    name: {
      color: '#666',
      margin: theme.spacing(1, 0),
      transition: 'all 0.3s ease-in',
    },
  }),
  { name: 'topPeople' },
);
