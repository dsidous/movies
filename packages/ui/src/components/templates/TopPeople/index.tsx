import React, { useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import TopPeopleProfile from '../../organisms/TopPeople';
import { Top_person } from '@typesRoots/toppeople';

interface Props {
  toppeople: Top_person[];
}

const TopPeople: React.FC<Props> = ({ toppeople }) => {
  const location = useLocation();
  const search = new URLSearchParams(location.search);
  const actPage = parseInt(search.get('page') || '1', 10);

  const history = useHistory();
  const [page, setPage] = useState(actPage);

  const handlePageSelect = (e: number) => {
    if (e > 0) {
      setPage(e);
      history.push(`/person?page=${e}`);
    }
  };

  return (
    <TopPeopleProfile
      page={page}
      toppeople={toppeople}
      handlePageSelect={handlePageSelect}
    />
  );
};

export default TopPeople;
