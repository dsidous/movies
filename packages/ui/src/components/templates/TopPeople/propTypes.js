import { bool, object, array } from 'prop-types';

export const contextTypes = {
  router: object.isRequired,
};

export const propTypes = {
  loading: bool,
  toppeople: array,
};

export const defaultProps = {};
