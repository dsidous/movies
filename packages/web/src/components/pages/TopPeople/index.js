import { compose, withProps } from 'recompose';
import { TopPeople } from '@movies/ui';
import { withTopPeople } from '@movies/common';

export default compose(
  withProps(props => {
    const params = new URLSearchParams(props.location.search);
    const page = parseInt(params.get('page'), 10) || 1;

    const query = `page=${page}`;
    return { query };
  }),
  withTopPeople(),
)(TopPeople);
