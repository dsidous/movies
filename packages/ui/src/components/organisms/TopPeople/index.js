import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import MyPager from '../../atoms/Pager';
import MediaImage from '../../atoms/MediaImage';
import PageTransition from '../../atoms/PageTransition/index';
import SEO from '../../atoms/SEO';

const TopPeopleProfile = ({ toppeople, page, handlePageSelect }) => (
  <PageTransition>
    <SEO title="Popular people" />
    <div className="top-people">
      <h2>POPULAR PEOPLE</h2>
      <div className="top-people__list">
        {toppeople.map(person => (
          <Link key={person.id} to={`/person/${person.id}`}>
            <figure className="top-people__list-element">
              <MediaImage
                mediaType="profile"
                size={1}
                filePath={person.profile_path}
                name={person.name}
                className="top-people__list-element__img"
              />
              <figcaption className="top-people__list-element__name">
                {person.name}
              </figcaption>
            </figure>
          </Link>
        ))}
      </div>
      <MyPager page={page} handlePageSelect={handlePageSelect} />
    </div>
  </PageTransition>
);

TopPeopleProfile.propTypes = propTypes;

export default TopPeopleProfile;
