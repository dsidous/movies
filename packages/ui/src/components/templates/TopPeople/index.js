import React, { useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import TopPeopleProfile from '../../organisms/TopPeople';

const TopPeople = ({ loading, toppeople }) => {
  const location = useLocation();
  const search = new URLSearchParams(location.search);
  const actPage = parseInt(search.get('page'), 10) || 1;

  const history = useHistory();
  const [page, setPage] = useState(actPage);

  const handlePageSelect = e => {
    if (e > 0) {
      setPage(e);
      history.push(`/person?page=${e}`);
    }
  };

  if (loading) {
    return <Spinner />;
  }

  return (
    <TopPeopleProfile
      page={page}
      toppeople={toppeople}
      handlePageSelect={handlePageSelect}
    />
  );
};

TopPeople.propTypes = propTypes;

export default TopPeople;
