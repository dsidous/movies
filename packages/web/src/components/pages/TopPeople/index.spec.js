import React from 'react';
import { shallow } from 'enzyme';
import TopPeople from '.';

const defaultProps = {
  location: { search: '1' },
};

describe('Pages/TopPeople', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<TopPeople {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
