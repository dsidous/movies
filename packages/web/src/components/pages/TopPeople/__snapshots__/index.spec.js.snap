// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/TopPeople should render as expected 1`] = `ShallowWrapper {}`;
